content/inside-rust/program-management-update-2025-08.md

```diff
@@ -0,0 +1,270 @@
+
+### RFC: pass pointers to `const`
+
+Rust has support for [inline assembly](https://doc.rust-lang.org/nightly/rust-by-example/unsafe/asm.html), via the `asm!` macro for example. Assembly is as close as one can get to writing machine code directly. This lets you do things the language doesn't have direct support such as accessing CPU registers or reading/writing values to device-specific segments in memory that allow control their behavior (this is often used when writing drivers).
+
+The macro lets you interpolate constant values into the assembly code you've written, similar to how you can interpolate values to a format string or macro:
+
+```rust
+println!("The meaning of life, universe and everything: {}", 42);
+```
+
+But constant values only get you so far. A common thing when writing assembly is to be able to pass *pointer* values (e.g. pointer to a specific field of a struct you want to manipulate) around. But currently, [only integer constant expressions are allowed](https://doc.rust-lang.org/1.89.0/reference/inline-assembly.html#r-asm.operand-type.supported-operands.sym).
+
+[Alice Ryhl](https://github.com/Darksonn) opened an RFC that [allows specifying pointers in the `const` operand](https://github.com/rust-lang/rfcs/pull/3848) too. Conceptually (and in assembly specifically) pointers could be thought of as just numbers that are interpreted as addresses to memory (although in Rust, the story is far more complex and [pointers are not the same thing as integers](https://doc.rust-lang.org/std/ptr/index.html#provenance)).
+
+This is now ready for feedback from the Lang team so I've opened a [design issue](https://github.com/rust-lang/lang-team/issues/347)  and got it scheduled for review and discussion.
+
+### Field projections
+
+[field-projections]: https://rust-lang.github.io/rust-project-goals/2025h2/field-projections.html
+
+We also had a design meeting on [Field Projections][field-projections]. When you have a type behind e.g. `Box` or `Rc`, you can access its field "directly" as if the wrapper/pointer type wasn't there:
+
+```rust
+struct Position {
+    x: f32,
+    y: f32,
+}
+
+fn main() {
+    let pos = Position{ x: 0.0, y: 0.0 };
+    let boxed = Box::new(pos);
+    println!("x: {}, y: {}", boxed.x, boxed.y);
+}
+```
+
+This is so common that we take it for granted, but when you write `boxed.x`, we first need to dereference `boxed` (follow the pointer to the `Position` struct in the memory) and then get its `x` field.
+
+And while this works for references and some pointer types, there's a long list of wrapper types where field access makes sense but it's not implemented because the semantics or limitations are different from the regular `Deref/DerefMut` traits. For example: `MaybeUninit<T>`, `Pin<T>`, `Cell<T>`, or the raw pointers `*const T`/`*mut T`. And of course custom types.
```

Comment:
I'd recommend using `Pin<P>` to differentiate it from the others that use `T` (since you always wrap a pointer type in `Pin` and never a `T` directly).